### DISCUSSION TITLE
How to implement passwordless authentication using Webauthn and Passkey for Flutter cross platform  native apps?

### DISCUSSION THREAD

**Discussion by thomastthai** [ory/kratos#2967] [Q&A]
Webauthn and PassKey can be used for web apps passwordless logins. How would passwordless authentication be implemented with native apps built using Flutter?

**Comment by vinckr** [comment]
Hello @thomastthai,

I suggest to take a look at how it is implemented in our reference [Node.js frontend](https://github.com/ory/kratos-selfservice-ui-node/
): 
- https://github.com/ory/kratos-selfservice-ui-node/blob/master/src/routes/sessions.ts
- https://github.com/ory/kratos-selfservice-ui-node/blob/master/src/routes/login.ts#L103
- https://github.com/ory/kratos-selfservice-ui-node/blob/master/src/pkg/middleware.ts#L17

Should be similar in the React Native app: https://github.com/ory/kratos-selfservice-ui-react-native

We have an official Flutter example here: https://www.ory.sh/docs/getting-started/integrate-auth/flutter-web-redirect, that is without MFA as far as I know.

**Reply by thomastthai** [reply]
@vinckr, thank you for the links. I've looked at those before. Perhaps I missed how they implemented Passkey natively.

**Reply by kmherrmann** [reply]
Please see here: https://github.com/ory/kratos/discussions/3038#discussioncomment-4755893 
The native mobile support for passkeys is still work in progress on our end.

**Reply by thomastthai** [reply]
@kmherrmann, thanks for the clarification. I'm looking forward to kratos' passkeys support for native mobile!

**Reply by thomastthai** [reply]
It looks like native support was announced for iOS. I didn't see any mention of Android or other platforms. Would you mind sharing an update?

**Comment by ouchi2501** [comment]
@kmherrmann 

Is this Discussion making progress?
I think passkeys are still not supported on native mobile at this point.
